### PR TITLE
temporary fix for jit test backward compatibility issues

### PR DESCRIPTION
**Pull request by albanD**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31949 temporary fix for jit test backward compatibility issues**

Differential Revision: [D19314763](https://our.internmc.facebook.com/intern/diff/D19314763)

### CODE REVIEW

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 0a7290be:


**None of the build failures appear to be your fault**.

* **1/1** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=c66ca74f03b6ff090142c34b602b5a4c45523d47&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base c66ca74f since Jan 08
  <details>
  <summary>
  You may want to rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=0a7290be2cf766ba1df9a60a855d6903709ce18a).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably **caused by upstream breakages**:
* [pytorch_linux_backward_compatibility_check_test](https://dr.pytorch.org/build-details.html?build_id=81955943) from Jan 08



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Review by houseroad — COMMENTED** [review]
The test is still broken
```
TypeError: append() takes exactly one argument (2 given)
```

So create a tuple? :-)

**Comment by albanD** [comment]
Yes, just saw the error...

File: test/backward_compatibility/check_backward_compatibility.py [hunk 1]
```diff
@@ -57,6 +57,19 @@
     ('aten::Int', datetime.date(2020, 1, 30)),
 ]
 
+jit_test_functions = [
+    '_TorchScriptTesting_StackString::pop',
```
Comment (houseroad) [inline test/backward_compatibility/check_backward_compatibility.py]: Nit: Also, just use `_TorchScriptTesting_` should be enough, and you can directly add this to the whitelist, and add a comment at the end.

Comment (albanD) [reply test/backward_compatibility/check_backward_compatibility.py]: I wanted to be as restrictive as possible to discourage people from adding more until we get a proper solution for the problem. But I dont have a very strong opinion. You think the more generic check is better?

Comment (houseroad) [reply test/backward_compatibility/check_backward_compatibility.py]: If so, we should add comments around `jit_test_functions` to warning the developers in future, and tell them don't add new testing functions. They will see this note when they try to add new items to white list in future.

**Review by houseroad — APPROVED** [review]
Unblock, please merge after the BC CI is green :-)

**Comment by bhosmer** [comment]
@albanD @houseroad sorry for the late comment, but - since these are exactly the signatures that are deliberately omitted when loading the new schemas from file ([here](https://github.com/pytorch/pytorch/blob/9a3cb1e859495477d95bf5fb5255e02fd4df82b5/test/backward_compatibility/check_backward_compatibility.py#L132)), I'm curious why we wouldn't just explicitly factor the `if "__torch__.torch.classes" in` filter out into a helper and call it from both places? 

No idea the history here, so apologies if the question is nonsense :)

**Comment by albanD** [comment]
@bhosmer I am not sure. The nice thing with the whitelist is that we can give a timeout to make sure the issue get solved.

**Comment by bhosmer** [comment]
@albanD yeah that's nice. I just hope the connection with the TODO logic in the new-schema file loader doesn't get lost. @houseroad that was a recent addition, right? Does that TODO have a definite future or do we risk having to rediscover the connection when the whitelist times these additions out?

**Comment by houseroad** [comment]
https://github.com/pytorch/pytorch/pull/31982 will remove `if "__torch__.torch.classes" in`, which was only for a temporary fix. The whitelist works after we load all the new functions. However, sometimes, we get problem to load schemas, `if "__torch__.torch.classes" in` is to bypass the problem.

**Comment by bhosmer** [comment]
@houseroad ah got it, thanks for the info!

**Pull request closed without merging**